vcx/libvcx/src/issuer_credential.rs

```diff
@@ -1041,4 +1079,224 @@ pub mod tests {
         let payment = serde_json::to_string(&credential.get_payment_txn().unwrap()).unwrap();
         assert!(payment.len() > 20);
     }
+
+    #[test]
+    fn test_encode_with_several_attributes_success() {
+
+        /*
+        for reference....expectation is encode_attributes returns this:
+
+        let expected = json!({
+          "address2": {
+            "encoded": "68086943237164982734333428280784300550565381723532936263016368251445461241953",
+            "raw": "101 Wilson Lane"
+          },
+          "zip": {
+            "encoded": "87121",
+            "raw": "87121"
+          },
+          "city": {
+            "encoded": "101327353979588246869873249766058188995681113722618593621043638294296500696424",
+            "raw": "SLC"
+          },
+          "address1": {
+            "encoded": "63690509275174663089934667471948380740244018358024875547775652380902762701972",
+            "raw": "101 Tela Lane"
+          },
+          "state": {
+            "encoded": "93856629670657830351991220989031130499313559332549427637940645777813964461231",
+            "raw": "UT"
+          }
+        });
+        */
+
+        static TEST_CREDENTIAL_DATA: &str =
+            r#"{"address2":["101 Wilson Lane"],
+            "zip":["87121"],
+            "state":["UT"],
+            "city":["SLC"],
+            "address1":["101 Tela Lane"]
+            }"#;
+
+        let results_json = encode_attributes(TEST_CREDENTIAL_DATA).unwrap();
+
+        let results : Value = serde_json::from_str(&results_json).unwrap();
+        let address2 : &Value = &results["address2"];
+
+        assert_eq!("68086943237164982734333428280784300550565381723532936263016368251445461241953", address2["encoded"]);
```

Comment:
Maybe better to use `encode` function to get expected values instead of hardcoded?